### ISSUE TITLE
Cannot build Sozi on Debian Stretch (64bit) with local Node/Npm

### ISSUE THREAD

**Issue by henrythebuilder** [senshu/Sozi#427]
This new issue is in response to the comment received by @senshu into pull request #425.
I'm not enable to build Sozi from source into my machine
I followed the instructions from 'README' but I could not complete the compilation.

### Summary of your problem or feature request
System configuration:
- Operating System: Debian GNU/Linux 9.4 (stretch) - 64bit
- Node/Npm: installed into home directory from pre-built package (Linux Binaries x64)
-- node -v >> v8.10.0 
-- npm -v >> 5.6.0 

### Version of Sozi and other relevant software
* Which version of Sozi are you using?
   Sozi source from master on github

* Which drawing software did you use? (examples: Inkscape 0.48, LibreOffice Draw 4.2)
  inkscape 0.92.1 (Debian package)

* On which platform? 
Debian GNU/Linux 9.4 (stretch) - 64bit

* If your problem happens when playing a presentation, which browsers have you tried? (examples: Firefox 46, Chrome 52)
Not a browser problem (I think) but generally I use chromium (Debian package:  64.0.3282.119-1~deb9u1)

### Steps to reproduce the problem
* Clone git repository (git clone git://github.com/senshu/Sozi.git)
* Install Droid Sans fonts copying ttf and eot files into vendor/DroidSans/ 
* From the root of the source tree execute `npm install`, `bower install` and  `grunt --config=config.linux-x64.json`

### Expected behavior

A successful build with results into a dist folder

### Observed behavior
Build stop with error
 
```
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN Invalid version: "18.01"
npm WARN Sozi No description
npm WARN Sozi No repository field.
npm WARN Sozi No README data
npm WARN Sozi No license field.

up to date in 0.115s
bower font-awesome#^4.7.0       cached https://github.com/FortAwesome/Font-Awesome.git#4.7.0
bower font-awesome#^4.7.0     validate 4.7.0 against https://github.com/FortAwesome/Font-Awesome.git#^4.7.0
bower font-awesome#^4.7.0      install font-awesome#4.7.0

font-awesome#4.7.0 bower_components/font-awesome
grunt-cli: The grunt command line interface (v1.2.0)

Fatal error: Unable to find local grunt.

If you're seeing this message, grunt hasn't been installed locally to
your project. For more information about installing and configuring grunt,
please see the Getting Started guide:

http://gruntjs.com/getting-started
```

### Hints and solutions (optional)
I also tried to manually install required packages:
* `npm install bower grunt-cli --save-dev`
* `npm install grunt --save-dev` 
 ... and so on for those gradually required (_nunjucks load-grunt-tasks grunt-babel grunt-browserify grunt-contrib-clean grunt-contrib-compress grunt-contrib-copy grunt-contrib-csslint grunt-contrib-jshint grunt-contrib-uglify grunt-electron grunt-install-dependencies grunt-jspot grunt-newer grunt-po2json grunt-rename grunt-rsync grunt-simple-mocha babel-core babel-preset-es2015_)

the build finish but nothing happens, in the sense that when I try to launch Sozi a window appears but is empty as shown in the attached image
Any help is welcome
Thanks in advance

![screenshot from 2018-03-18 15-16-31](https://user-images.githubusercontent.com/6670654/37566896-7a8d1866-2abf-11e8-90ca-771b12228ee6.png)

**Comment by senshu** [comment]
This may be related to the version of npm that you are using: I have only tested with npm 4.

**Comment by senshu** [comment]
However, if the build was successful, Sozi should work correctly.
I'm not sure that the empty UI is related to the build issue.

**Comment by henrythebuilder** [comment]
Ok, thanks for the support. 
I'll try again some other attempt ...

**Comment by henrythebuilder** [comment]
updating ...
I made a build with additional command from scratch capturing output and verifying the differences from the pre-released package  (Sozi 18.03) 
Into the build packages some files are missing including the entire node_modules directory.
Copying missing files to the 'build' directory Sozi seem to work.

Build command:
```
npm install bower grunt-cli
npm install grunt
npm install nunjucks
npm install load-grunt-tasks
npm install grunt-babel grunt-browserify grunt-contrib-clean grunt-contrib-compress grunt-contrib-copy grunt-contrib-csslint grunt-contrib-jshint grunt-contrib-uglify grunt-electron grunt-install-dependencies grunt-jspot grunt-newer grunt-po2json grunt-rename grunt-rsync grunt-simple-mocha
npm install babel-core
npm install babel-preset-es2015
npm install
bower install
grunt -v --config=config.linux-x64.json
```
I attach the log file and the report of the differences if they can help to understand the problem.
I apologize for my little knowledge of node/npm


![sozi-build-diff-from-latest-release](https://user-images.githubusercontent.com/6670654/37569803-d89c28d4-2ae7-11e8-99b8-00ac3501ddd4.png)
[sozi-build-20180318.log](https://github.com/senshu/Sozi/files/1823023/sozi-build-20180318.log)

**Comment by alexisruiz96** [comment]
Hi, i'm facing the same problem and i'm using the version 4.1.2 of npm. Also i have the version 7.7.4 of nodejs. Which version are you using to obtain a good result @senshu ? Or @henrythebuilder , did you solve the problem? Because i've done the same steps as you and i don't get a good result.

**Comment by henrythebuilder** [comment]
Hi, I tried to compile the last version of Sozi with nodejs 7.10.1 and 8.11.2 (LTS) installed locally, the build proceeds to the end using the 'github' indications but unfortunately I got the same result: when I try to launch Sozi a window appears but is empty as in the previous message.
Also in this case comparing the files with pre-released package some files are missing.
Note that node_modules in these cases is empty (resources/app/node_modules) and and trying to copy node_modules files from the root of the source tree into the 'dist' output then Sozi seem to work: window is not empty and  the 'choose file dialog' appears.
